src/scripts/create_schibsted.py

```diff
@@ -0,0 +1,110 @@
+"""Create the Schibsted summarisation dataset."""
+
+import fasttext
+import pandas as pd
+from constants import MAX_NUM_CHARS_IN_ARTICLE, MIN_NUM_CHARS_IN_ARTICLE
+from datasets import Dataset, DatasetDict, Split, load_dataset
+from huggingface_hub import HfApi, hf_hub_download
+from requests import HTTPError
+
+
+def main():
+    """Create the Schibsted summarisation dataset and upload to HF Hub."""
+    dataset_id = "Schibsted/schibsted-article-summaries"
+
+    dataset = load_dataset(dataset_id, token=True)
+    assert isinstance(dataset, DatasetDict)
+
+    # Rename validation split to val
+    dataset["val"] = dataset.pop("validation")
+
+    # Rename article_text_all to text
+    dataset = dataset.rename_columns(
+        column_mapping=dict(article_text_all="text", summary="target_text")
+    )
+
+    # Add article title to text, separated by a colon
+    dataset = dataset.map(lambda x: {"text": f"{x['article_title']}: {x['text']}"})
+
+    # Summary is a list of strings, but should be a single string
+    dataset = dataset.map(lambda x: {"target_text": " ".join(x["target_text"])})
+
+    # Ignore samples outside the bounds
+    train_df = dataset["train"].to_pandas()
+    val_df = dataset["val"].to_pandas()
+    test_df = dataset["test"].to_pandas()
+    assert isinstance(train_df, pd.DataFrame)
+    assert isinstance(val_df, pd.DataFrame)
+    assert isinstance(test_df, pd.DataFrame)
+
+    # Only work with samples where the text is not very large or small
+    train_lengths = train_df.text.str.len()
+    val_lengths = val_df.text.str.len()
+    test_lengths = test_df.text.str.len()
+    lower_bound = MIN_NUM_CHARS_IN_ARTICLE
+    upper_bound = MAX_NUM_CHARS_IN_ARTICLE
+    train_df = train_df[train_lengths.between(lower_bound, upper_bound)]
+    val_df = val_df[val_lengths.between(lower_bound, upper_bound)]
+    test_df = test_df[test_lengths.between(lower_bound, upper_bound)]
+
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Dataset is a mix of Swedish and Norwegian articles.
+    # Make two separate datasets, one for each language.
+    model_path = hf_hub_download(
+        repo_id="facebook/fasttext-language-identification", filename="model.bin"
+    )
+    model = fasttext.load_model(model_path)
+
+    def detect_language(text: str) -> str:
+        """Detect if the text is in Swedish or Norwegian.
+
+        Args:
+            text (str):
+                The text to detect the language of.
+
+        Returns:
+            str:
+                The language of the text ("sv" or "no").
```

Comment:
Same thing as above, since the return type is declared in the function type:
```suggestion
            The language of the text ("sv" or "no").
```